File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int max_v = 0;
        int max_h = 0;
        for (int[] size : sizes) {
            int v = Math.max(size[0], size[1]);
            int h = Math.min(size[0], size[1]);
            max_v = Math.max(max_v, v);
            max_h = Math.max(max_h, h);
        }
        return max_v * max_h;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.swift
import Foundation

func solution(_ sizes:[[Int]]) -> Int {
    var w = 0
    var h = 0
    for size in sizes {
        w = max(w, max(size[0], size[1]))
        h = max(h, min(size[0], size[1]))
    }
    
    return w * h
}